### PR TITLE
Restrict installation logging to workflow srcdir

### LINKED ISSUE #4220: log version control: diff from outside the workflow source dir listed
The `log/version/` files log changes for files that aren't in the workflow source dir.

For example if we have this directory structure:

```
myproject/
    src/
    ...
    test-workflow/
        bin/
        flow.cylc
        ...
    build/workflow/
        flow.cylc
        ...
```

And we install like so:

```console
$ cylc install -c 'myproject/test-workflow'
```

We could see uncommitted diff from `myproject/build-workflow` listed in the log files. This is the case for `git` vc, haven't tested `svn`.

The `rose-stem` pattern often sees test workflows embedded in much larger projects, listing the diff of the whole project would be over the top for Cylc purposes. It could be useful for provenance, however, Cylc doesn't capture full information about its run environment so should this be captured via other means?

**Question:**

* Should we restrict this to just show changes from within the workflow source directory? - **Yes**

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by wxtim**
These changes close #4220 
This is a small change with no associated Issue.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.py` and
  `conda-environment.yml`.
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No documentation PR Req'd: This feature is documented in the plugin module docstring which is ingested by cylc-doc. I have not changed this, because this change makes the description less misleading.

### CODE REVIEW

File: cylc/flow/install_plugins/log_vc_info.py [hunk 1]
```diff
@@ -268,9 +268,10 @@ def get_diff(vcs: str, path: Union[Path, str]) -> Optional[str]:
         vcs: The version control system.
         path: The path to the repo.
     """
-    args = DIFF_COMMANDS[vcs]
+    args_ = DIFF_COMMANDS[vcs]
```
Comment (wxtim) [inline cylc/flow/install_plugins/log_vc_info.py:271]: `args` (rather than `args_`) make using pdb tricky.

**Comment by hjoliver** [comment]
@wxtim your new tests pass locally for me, but @oliver-sanders's example on #4220 fails with this diff in `myproject`:
```diff
┬─[oliverh@niwa-1007885:~/c/myproject]─[12:31:09]─[I]─[G:master]
╰─>$ pwd
/home/oliverh/cylc-src/myproject
┬─[oliverh@niwa-1007885:~/c/myproject]─[12:31:11]─[I]─[G:master]
╰─>$ git diff
diff --git a/build/workflow/flow.cylc b/build/workflow/flow.cylc
index e69de29..907b308 100644
--- a/build/workflow/flow.cylc
+++ b/build/workflow/flow.cylc
@@ -0,0 +1 @@
+blah
diff --git a/src/a b/src/a
index e69de29..907b308 100644
--- a/src/a
+++ b/src/a
@@ -0,0 +1 @@
+blah
diff --git a/test-workflow/flow.cylc b/test-workflow/flow.cylc
index e69de29..907b308 100644
--- a/test-workflow/flow.cylc
+++ b/test-workflow/flow.cylc
@@ -0,0 +1 @@
+blah
```
Result:
```
┬─[oliverh@niwa-1007885:~/cylc-src]─[12:33:06]─[N]
╰─>$ cylc install -C myproject/test-workflow
INSTALLED test-workflow/run1 from myproject/test-workflow
PluginError: Error in plugin cylc.post_install.log_vc_info
[Errno 128] fatal: ambiguous argument 'myproject/test-workflow': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

It works on master, albeit with the out-of-src-dir diffs logged.

**Comment by hjoliver** [comment]
Looks like your change assumes an absolute source path on the `cylc install` command line.

**Comment by wxtim** [comment]
> Looks like your change assumes an absolute source path on the `cylc install` command line.

I think that I have fixed this.

**Review by kinow — APPROVED** [review]
Using `master`, after re-creating the scenario described in the issue linked (also `git add` and commit, then modify a file), `uncommitted.diff`:

```bash
# Auto-generated diff of uncommitted changes in the Cylc workflow repository:
#   myproject/test-workflow
diff --git a/build/workflow/flow.cylc b/build/workflow/flow.cylc
index 56f0067..0275ba1 100644
--- a/build/workflow/flow.cylc
+++ b/build/workflow/flow.cylc
@@ -1,3 +1,7 @@
+
+
+
+
 [scheduling]
   cycling mode = integer
   initial cycle point = 1
```

Using this branch, `uncommitted.diff`:

```
# Auto-generated diff of uncommitted changes in the Cylc workflow repository:
#   myproject/test-workflow
```

So not displaying the changes in the other directory :tada:  ! `vcs.conf` still shows the changes, but I think that's OK:

```
version control system = "git"
repository version = "81f03d9-dirty"
commit = "81f03d9c7c83399a44216e8db3272b7475a46afa"
working copy root path = "myproject/test-workflow"
status = """
 M ../build/workflow/flow.cylc
"""
```

+1

**Review by hjoliver — APPROVED** [review]
Noice :100:

**Pull request merged**